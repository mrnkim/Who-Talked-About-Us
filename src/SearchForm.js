import { React, useState } from "react";
import InputForm from "./InputForm";
import TwelveLabsApi from "./api";

function SearchForm({ index, search }) {
  const [query, setQuery] = useState("");
  const [error, setError] = useState("");
  const [showComponents, setShowComponents] = useState(false);

  /** Update form input. */
  function handleChange(evt) {
    const input = evt.target;
    setQuery(input.value);
  }

  /** Call parent function and clear form. */
  async function handleSubmit(evt) {
    evt.preventDefault();
    const trimmedQuery = query.trim();
    const response = await search(index, trimmedQuery);
  }
  return (
    <div>
      <InputForm
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        input={query}
        type="search"
        buttonText="Search"
      />
    </div>
  );
}

export default SearchForm;
