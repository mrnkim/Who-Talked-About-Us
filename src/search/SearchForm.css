.errorMessageSearch {
    color: var(--TL-Red-500, #F44336);
    font-size: 0.875rem;
    font-style: normal;
    font-weight: 400;
    line-height: 1.25rem;
}

.inputAndErrorSearch{
    gap: 0.25rem;
    width: 100%;
}

