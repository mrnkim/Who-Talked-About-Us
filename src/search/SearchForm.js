import React, { useState } from "react";
import sanitize from "sanitize-filename";
import InputForm from "../common/InputForm";
import searchIcon from "../svg/Union.svg";
import "./SearchForm.css";

/** Form to search videos
 *
 * VideoComponents -> SearchForm -> InputForm
 *
 */

function SearchForm({ searchQuery, setSearchQuery, setFinalSearchQuery }) {
  const [error, setError] = useState("");

  function handleChange(evt) {
    const input = evt.target;
    setSearchQuery(input.value);
    if (error && input.value.trim() !== "") {
      setError("");
    }
  }

  function handleSubmit(evt) {
    evt.preventDefault();
    if (searchQuery) {
      const sanitizedQuery = sanitize(searchQuery);
      if (!sanitizedQuery) {
        setError("Please enter the search term");
      } else {
        try {
          setFinalSearchQuery(sanitizedQuery);
        } catch (error) {
          console.error(error);
        }
      }
    }
  }

  return (
    <div className="inputAndErrorSearch">
      <InputForm
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        type="What are you looking for? (e.g., applying MAC gold highlighter)"
        icon={searchIcon}
        value={searchQuery}
        className="searchTextField"
      />
      {error && <div className="errorMessageSearch">{error}</div>}
    </div>
  );
}

export default SearchForm;
