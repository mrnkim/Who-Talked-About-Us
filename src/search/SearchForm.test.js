import React from "react";
import { render, fireEvent, waitFor } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import SearchForm from "./SearchForm";

jest.mock("../api/apiHooks");

describe("SearchForm", () => {
  it("renders without crashing", () => {
    const { getByLabelText } = render(<SearchForm />);
    expect(getByLabelText(/What are you looking for?/i)).toBeInTheDocument();
  });
});
