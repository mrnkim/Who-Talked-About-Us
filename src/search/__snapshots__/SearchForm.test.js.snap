// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div>
    <div
      class="mt-5 container"
    >
      <form
        class="d-flex justify-content-center align-items-center"
      >
        <input
          aria-label="What are you looking for? (e.g., applying MAC gold highlighter)"
          class="form-control"
          data-testid="search-input"
          id="What are you looking for? (e.g., applying MAC gold highlighter)"
          name="What are you looking for? (e.g., applying MAC gold highlighter)"
          placeholder="What are you looking for? (e.g., applying MAC gold highlighter)"
          style="width: 60%; font-size: 1em;"
          type="What are you looking for? (e.g., applying MAC gold highlighter)"
          value=""
        />
        <button
          class="rounded btn btn-primary"
          style="margin-left: 3px;"
          type="submit"
        >
          <i
            class="bi bi-search"
          />
           Search
        </button>
      </form>
    </div>
  </div>
</div>
`;
